tests/test_builders/test_build_linkcheck.py

```diff
@@ -900,39 +909,39 @@ class FakeResponse:
     url = "http://localhost/"
 
 
-def test_limit_rate_default_sleep(app):
+def test_limit_rate_default_sleep(app: Sphinx) -> None:
     worker = HyperlinkAvailabilityCheckWorker(app.config, Queue(), Queue(), {})
     with mock.patch('time.time', return_value=0.0):
         next_check = worker.limit_rate(FakeResponse.url, FakeResponse.headers.get("Retry-After"))
     assert next_check == 60.0
 
 
-def test_limit_rate_user_max_delay(app):
-    app.config.linkcheck_rate_limit_timeout = 0.0
+def test_limit_rate_user_max_delay(app: Sphinx) -> None:
+    app.config.linkcheck_rate_limit_timeout = 0.0  # type: ignore[attr-defined]
```

Comment:
why have you used `confoverrides` above but not here?

Reply:
The truth is that I've read and re-read most of the tests in this file, but these few here -- the ones that don't involve test HTTP servers -- remain a bit of a mystery to me.  That means I'm less inclined to make multi-line or multi-word edits; I'm trying to use the smallest change possible because I don't feel like a maintainer of these.

Reply:
Those tests are actually using the `html` builder and the `test-root` but actually, we are using an `app` because we just want to get its default configuration... 

I feel that we need a fixture with a simple configuration instead.

Reply:
Ok; recently I was reconsidering the choice _not_ to move this into `confoverrides`  it seems like there's more consensus that it would make sense to use `confoverrides`, so I'll push that change here soon.

Reply:
Oh no, for this one it's fine. What I meant is that we should not need to have a `app` to get a configuration but be able to have a standalone `config` object.